src/diffusers/pipelines/animatediff/pipeline_animatediff_video2video.py

```diff
@@ -0,0 +1,916 @@
+            )
+
+        if latents is None:
+            video = video.to(device=device, dtype=dtype)
+
+            # make sure the VAE is in float32 mode, as it overflows in float16
+            if self.vae.config.force_upcast:
+                video = video.float()
+                self.vae.to(dtype=torch.float32)
+
+            if isinstance(generator, list):
+                if len(generator) != batch_size:
+                    raise ValueError(
+                        f"You have passed a list of generators of length {len(generator)}, but requested an effective batch"
+                        f" size of {batch_size}. Make sure the batch size matches the length of the generators."
+                    )
+
+                init_latents = [
+                    retrieve_latents(self.vae.encode(video[i]), generator=generator[i]) for i in range(batch_size)
+                ]
+                init_latents = torch.stack(init_latents)
+            else:
+                init_latents = torch.stack(
+                    [retrieve_latents(self.vae.encode(vid), generator=generator) for vid in video]
+                )
+
+            # restore vae to original dtype
+            if self.vae.config.force_upcast:
+                self.vae.to(dtype)
+
+            init_latents = init_latents.to(dtype)
+            init_latents = self.vae.config.scaling_factor * init_latents
+
+            if batch_size > init_latents.shape[0] and batch_size % init_latents.shape[0] == 0:
+                # expand init_latents for batch_size
+                deprecation_message = (
+                    f"You have passed {batch_size} text prompts (`prompt`), but only {init_latents.shape[0]} initial"
+                    " images (`image`). Initial images are now duplicating to match the number of text prompts. Note"
+                    " that this behavior is deprecated and will be removed in a version 1.0.0. Please make sure to update"
+                    " your script to pass as many initial images as text prompts to suppress this warning."
+                )
+                deprecate("len(prompt) != len(video)", "1.0.0", deprecation_message, standard_warn=False)
+                additional_video_per_prompt = batch_size // init_latents.shape[0]
+                init_latents = torch.cat([init_latents] * additional_video_per_prompt, dim=0)
+            elif batch_size > init_latents.shape[0] and batch_size % init_latents.shape[0] != 0:
+                raise ValueError(
+                    f"Cannot duplicate `image` of batch size {init_latents.shape[0]} to {batch_size} text prompts."
+                )
+            else:
+                init_latents = torch.cat([init_latents], dim=0)
+
+            noise = randn_tensor(init_latents.shape, generator=generator, device=device, dtype=dtype)
+            latents = self.scheduler.add_noise(init_latents, noise, timestep).permute(0, 2, 1, 3, 4)
+        else:
+            if latents.dim() != 5:
```

Comment:
I would define a `shape` variable with the expected latents shape, like it's done here 
https://github.com/huggingface/diffusers/blob/2e4dc3e25d6abaf5f15d7e5c62db75a8bd63f915/src/diffusers/pipelines/text_to_video_synthesis/pipeline_text_to_video_synth.py#L471

And check that the passed in latents shape matches.